### PR TITLE
SPOI-11557 Added metrics support for Jdbc poller and output operator.

### PR DESCRIPTION
**Pull request by deepak-narkhede**
1) Extension of Jdbc poller and output operators with added metrics.
2) Changes to Database to database sync app to reflect extended operators.

Conflicts:
	app-templates/pom.xml

### CODE REVIEW

**Comment by deepak-narkhede** [comment]
@yogidevendra . Please review.

**Comment by deepak-narkhede** [comment]
@chaithu14 . kindly review.

File: app-templates/database-to-database-sync/pom.xml [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
   <modelVersion>4.0.0</modelVersion>
-  
-  <groupId>com.datatorrent.apps</groupId>
+  <parent>
+    <groupId>com.datatorrent</groupId>
+    <artifactId>moodi-app-templates</artifactId>
+    <version>0.1.0-SNAPSHOT</version>
+  </parent>
+  <groupId>com.datatorrent</groupId>
   <version>0.9.0</version>
```
Comment (chaithu14) [inline app-templates/database-to-database-sync/pom.xml]: Change the version to 0.10.0

File: app-templates/database-to-database-sync/pom.xml [hunk 2]
```diff
@@ -13,9 +17,9 @@
 
   <properties>
     <!-- change this if you desire to use a different version of Apex Core -->
-    <apex.version>3.4.0</apex.version>
+    <apex.version>3.6.0</apex.version>
```
Comment (chaithu14) [inline app-templates/database-to-database-sync/pom.xml]: Change the apex version to 3.7.0-SNAPSHOT.

File: app-templates/database-to-database-sync/pom.xml [hunk 3]
```diff
@@ -311,6 +315,21 @@
       <artifactId>commons-dbutils</artifactId>
       <version>1.6</version>
     </dependency>
+    <dependency>
+      <groupId>com.datatorrent</groupId>
+      <artifactId>moodi-operator-library</artifactId>
+      <version>0.1.0-SNAPSHOT</version>
```
Comment (chaithu14) [inline app-templates/database-to-database-sync/pom.xml]: This dependency is it in parent pom, so, please remove it.

File: operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOPollInputOperator.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ */
+public class JdbcPOJOPollInputOperator extends com.datatorrent.lib.db.jdbc.JdbcPOJOPollInputOperator
+{
+  @AutoMetric
+  private long totalEventsRead = 0;
+  @AutoMetric
+  private long eventsReadPerSecond;
+  private long eventsReadPerWindow;
+  private double windowTimeSec;
```
Comment (chaithu14) [inline operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOPollInputOperator.java]: Make eventsReadPerSecond, eventsReadPerWindow, windowTimeSec variables as transient.

File: operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOInsertOutputOperator.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+public class JdbcPOJOInsertOutputOperator extends com.datatorrent.lib.db.jdbc.JdbcPOJOInsertOutputOperator
+{
+  @AutoMetric
+  private long totalEventsWritten = 0;
+  @AutoMetric
+  private long eventsWrittenPerSecond;
+  @AutoMetric
+  private long totalBadEvents = 0;
+  private long eventsWrittenPerWindow;
+  private double windowTimeSec;
```
Comment (chaithu14) [inline operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOInsertOutputOperator.java]: Make eventsReadPerSecond, eventsReadPerWindow, windowTimeSec variables as transient.

File: app-templates/database-to-database-sync/src/main/resources/resources/ui/dashboards/Database-to-Database-Sync.dtdashboard [hunk 6]
```diff
@@ -0,0 +1 @@
...ully","id":"tuplesWrittenSuccessfully","type":"long","sort":"string","filter":"like"}],"fieldSelectionOptions":{"initialSorts":[{"id":"id","dir":"+"}],"bgSizeMultiplier":"2","defaultRowHeight":"29","loading":false},"fieldSelectionColumns":[{"id":"selector","key":"id","label":"","selector":true,"width":"40px","lockWidth":true},{"id":"id","key":"id","sort":"string","filter":"like"},{"id":"type","key":"type","sort":"string","filter":"like"}]},"dataQueryParams":{"data":{"fields":["eventsWrittenPerSecond","totalEventsWritten","errorTuples","totalBadEvents","tuplesWrittenSuccessfully"],"schemaKeys":{"appName":"Database-to-Database-Sync","appUser":"deepak","logicalOperatorName":"JdbcOutput"}},"incompleteResultOK":true,"getDataContinuously":true},"chart":{"keyField":"eventsWrittenPerSecond","valueFields":["eventsWrittenPerSecond","totalEventsWritten","errorTuples","totalBadEvents","tuplesWrittenSuccessfully"],"valueField":"totalBadEvents"},"sortOptions":{"enabled":false,"key":{"name":"eventsWrittenPerSecond","type":"long"},"direction":"asc"},"loadingData":false}}]}},"id":"deepak:Database-to-Database-Sync","description":"automatically generated dashboard","mtime":"2017-06-26T08:59:00.401Z"}
```
Comment (chaithu14) [inline app-templates/database-to-database-sync/src/main/resources/resources/ui/dashboards/Database-to-Database-Sync.dtdashboard]: Please format this json, which should be in readable format.

File: operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOPollInputOperator.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
+package com.datatorrent.db.jdbc;
+
```
Comment (chaithu14) [inline operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOPollInputOperator.java]: Move these operators to com.datatorrent.moodi.lib.db package.

File: app-templates/database-to-database-sync/pom.xml (hunk 1 above)
Comment (deepak-narkhede) [inline app-templates/database-to-database-sync/pom.xml]: Done.

File: app-templates/database-to-database-sync/pom.xml (hunk 2 above)
Comment (deepak-narkhede) [inline app-templates/database-to-database-sync/pom.xml]: Done.

File: app-templates/database-to-database-sync/pom.xml (hunk 3 above)
Comment (deepak-narkhede) [inline app-templates/database-to-database-sync/pom.xml]: Done.

File: operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOInsertOutputOperator.java (hunk 5 above)
Comment (deepak-narkhede) [inline operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOInsertOutputOperator.java]: Done.

File: app-templates/database-to-database-sync/src/main/resources/resources/ui/dashboards/Database-to-Database-Sync.dtdashboard (hunk 6 above)
Comment (deepak-narkhede) [inline app-templates/database-to-database-sync/src/main/resources/resources/ui/dashboards/Database-to-Database-Sync.dtdashboard]: Done.

File: operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOPollInputOperator.java (hunk 4 above)
Comment (deepak-narkhede) [inline operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOPollInputOperator.java]: Done.

File: operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOPollInputOperator.java (hunk 7 above)
Comment (deepak-narkhede) [inline operators/library/src/main/java/com/datatorrent/db/jdbc/JdbcPOJOPollInputOperator.java]: Done.

File: app-templates/database-to-database-sync/pom.xml (hunk 2 above)
Comment (chaithu14) [inline app-templates/database-to-database-sync/pom.xml]: Changes are not reflected. Could you please check.

File: app-templates/database-to-database-sync/pom.xml [hunk 8]
```diff
@@ -311,6 +315,16 @@
       <artifactId>commons-dbutils</artifactId>
       <version>1.6</version>
     </dependency>
+   <dependency>
+      <groupId>com.datatorrent</groupId>
+      <artifactId>metrics-api</artifactId>
+      <version>3.9.0-SNAPSHOT</version>
+    </dependency>
```
Comment (chaithu14) [inline app-templates/database-to-database-sync/pom.xml]: I think, metrics-api is not needed. Please have a look other app templates.

File: app-templates/database-to-database-sync/pom.xml [hunk 9]
```diff
@@ -13,9 +17,9 @@
 
   <properties>
     <!-- change this if you desire to use a different version of Apex Core -->
-    <apex.version>3.4.0</apex.version>
+    <apex.version>3.6.0</apex.version>
     <apex.apppackage.classpath>lib/*.jar</apex.apppackage.classpath>
-    <malhar.version>3.6.0</malhar.version>
+    <malhar.version>3.7.0-SNAPSHOT</malhar.version>
```
Comment (chaithu14) [inline app-templates/database-to-database-sync/pom.xml]: Apex and Malhar versions are there in parent pom. Please check it, if it is different version, then only add it here.

File: app-templates/database-to-database-sync/pom.xml [hunk 10]
```diff
@@ -311,6 +315,16 @@
...
+   <dependency>
+      <groupId>com.datatorrent</groupId>
+      <artifactId>metrics-api</artifactId>
+      <version>3.9.0-SNAPSHOT</version>
+    </dependency>
+    <dependency>
+      <groupId>com.datatorrent</groupId>
+      <artifactId>metrics-plugin</artifactId>
+      <version>3.9.0-SNAPSHOT</version>
+    </dependency>
```
Comment (chaithu14) [inline app-templates/database-to-database-sync/pom.xml]: Please add this plugin in parent pom.

**Comment by chaithu14** [comment]
LGTM

**Pull request merged**